tools/msu_finder.rb

```diff
@@ -0,0 +1,760 @@
+#!/usr/bin/env ruby
+
+###
+#
+# This sceript will enumerate download links for Microsoft patches.
+#
+# Author:
+# * sinn3r
+#
+###
+
+
+msfbase = __FILE__
+while File.symlink?(msfbase)
+  msfbase = File.expand_path(File.readlink(msfbase), File.dirname(msfbase))
+end
+$:.unshift(File.expand_path(File.join(File.dirname(msfbase), '..', 'lib')))
+require 'rex'
+require 'nokogiri'
+require 'uri'
+require 'json'
+require 'optparse'
+
+module MicrosoftPatch
+
+  module SiteInfo
+    TECHNET = {
+      ip:    '157.56.148.23',
+      vhost: 'technet.microsoft.com'
+    }
+
+    MICROSOFT = {
+      ip:    '104.72.230.162',
+      vhost: 'www.microsoft.com'
+    }
+
+    GOOGLEAPIS = {
+      ip:    '74.125.28.95',
+      vhost: 'www.googleapis.com'
+    }
+  end
+
+  # This class provides whatever other classes need.
+  class Base
```

Comment:
This base class is containing unrelated methods like "print" methods and then a method to send http requests. Even if you keep these unrelated methods together I'd say it looks like a good candidate for a mixin instead of using class inheritance.